### PR TITLE
rule(falco_rules): Polkit Local Privilege Escalation Vulnerability (CVE-2021-4034)

### PR DESCRIPTION
**Pull request by darryk10**
Polkit pkexec Local Privilege Escalation Vulnerability (CVE-2021-4034)

Co-authored-by: javery-sysdig <jason.avery@sysdig.com>

Signed-off-by: darryk10 <stefano.chierici@sysdig.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

/kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

/area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:
This rule detects an attempt to exploit a privilege escalation vulnerability in Polkit's pkexec. By running specially crafted code, a local user can leverage this flaw to gain root privileges on a compromised system. Reference: https://seclists.org/oss-sec/2022/q1/80
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
rule(Polkit Local Privilege Escalation Vulnerability): new rule created to detect CVE-2021-4034
```

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -3120,6 +3120,17 @@
   priority: NOTICE
   tags: [network, process, mitre_command_and_control]
 
+# This rule helps detect CVE-2021-4034:
+# A privilege escalation to root through memory corruption
+- rule: Polkit Local Privilege Escalation Vulnerability (CVE-2021-4034)
+  desc: "This rule detects an attempt to exploit a privilege escalation vulnerability in Polkit's pkexec. By running specially crafted code, a local user can leverage this flaw to gain root privileges on a compromised system"
+  condition:
+    spawned_process and user.uid != 0 and proc.name=pkexec and proc.args = '' and proc.env icontains 'GCONV_PATH'
```
Comment (jhoos) [inline rules/falco_rules.yaml:3128]: Why `GCONV_PATH` and not just `PATH`? The way the original vulnerability was described (https://blog.qualys.com/vulnerabilities-threat-research/2022/01/25/pwnkit-local-privilege-escalation-vulnerability-discovered-in-polkits-pkexec-cve-2021-4034) it was a manipulation of `PATH`. I wonder if we even need to check `proc.env` at all; it seems that any invocation of `pkexec` with an empty set of args would be suspect, no? Or am I misunderstanding what this all means to falco?

Comment (darryk10) [reply rules/falco_rules.yaml:3128]: Hey @jhoos, thanks for the feedback. 
I added this condition due to the exploitability and GCONV_PATH can force iconv_open() to read another configuration file. Ee-introducing GCONV_PATH into pkexec's environment, and to execute our own shared library it's possible to get root. and exploit CVE-2021-4034.
Reference: https://www.qualys.com/2022/01/25/cve-2021-4034/pwnkit.txt

Comment (Kaizhe) [reply rules/falco_rules.yaml:3128]: Jason, I tend to agree with you here. The exploit relies on a malformed environment variables which is hard to detect during the process execution (no baseline could compare). And execute `pkexe` may be a normal case which may give you bunch of options (the same behavior as `-h` or `--help`). If what I assumed before is correct, I don't see there is an easy to write a rule to detect such exploit.

Comment (darryk10) [reply rules/falco_rules.yaml]: Thanks @Kaizhe for your feedback. I changed the condition removing the proc.env check.
Thanks again.

**Review by Kaizhe — APPROVED** [review]

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -3120,6 +3120,17 @@
...
   tags: [network, process, mitre_command_and_control]
 
+# This rule helps detect CVE-2021-4034:
+# A privilege escalation to root through memory corruption
+- rule: Polkit Local Privilege Escalation Vulnerability (CVE-2021-4034)
+  desc: "This rule detects an attempt to exploit a privilege escalation vulnerability in Polkit's pkexec. By running specially crafted code, a local user can leverage this flaw to gain root privileges on a compromised system"
+  condition:
+    spawned_process and user.uid != 0 and proc.name=pkexec and proc.args = ''
+  output:
+    "Detect Polkit pkexec Local Privilege Escalation Exploit (CVE-2021-4034) (user=%user.loginname uid=%user.loginuid command=%proc.cmdline args=%evt.args)"
```
Comment (leogr) [inline rules/falco_rules.yaml:3130]:
```suggestion
    "Detect Polkit pkexec Local Privilege Escalation Exploit (CVE-2021-4034) (user=%user.loginname uid=%user.loginuid command=%proc.cmdline args=%proc.args)"
```

I guess you wanted to print the proc's arguments, right?

**Review by leogr — COMMENTED** [review]
I've found an issue in the output, otherwise SGMT.

I'm also a bit concerned about false positives since this rule triggers just every time `pkexec` is called without args.

File: rules/falco_rules.yaml (hunk 2 above)
Comment (anthony-alessi-acquia) [reply rules/falco_rules.yaml]: I'm a bit confused whether 'args=%proc.args' would output anything useful. Wouldn't it just be an empty string? Should we try to print out the environment variable that is being leveraged?

**Comment by Kaizhe** [comment]
> I've found an issue in the output, otherwise SGMT.
> 
> I'm also a bit concerned about false positives since this rule triggers just every time `pkexec` is called without args.

According to the fix, the `pkexe` will return error if runs without argument.

**Review by leogr — APPROVED** [review]
/approve

**Pull request merged**